log/logger.ts

```diff
@@ -179,11 +179,21 @@ export class Logger {
     return this.#_log(LogLevels.INFO, msg, ...args);
   }
 
+  /** @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead. */
   warning<T>(msg: () => T, ...args: unknown[]): T | undefined;
   warning<T>(msg: T extends GenericFunction ? never : T, ...args: unknown[]): T;
```

Comment:
```suggestion
  /** @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead. */
  warning<T>(msg: T extends GenericFunction ? never : T, ...args: unknown[]): T;
```
So both overloads show as deprecated.